chcemvediet/apps/inforequests/tests/test_models/test_deadline.py

```diff
@@ -0,0 +1,157 @@
+                (Deadline.UNITS.WORKDAYS,      False,         True),
+        )
+        # Make sure we are testing all defined deadline units
+        tested_deadline_units = [a for a, _, _ in tests]
+        defined_deadline_units = vars(Deadline.UNITS).values()
+        self.assertItemsEqual(tested_deadline_units, defined_deadline_units)
+
+        for deadline_unit, is_in_calendar_days, is_in_workdays in tests:
+            deadline = self._create_deadline(unit=deadline_unit)
+            self.assertEqual(deadline.is_in_calendar_days, is_in_calendar_days)
+            self.assertEqual(deadline.is_in_workdays, is_in_workdays)
+
+    def test_deadline_date_property(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-10'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS)
+        self.assertEqual(deadline.deadline_date, naive_date(u'2010-10-18'))
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-10'), value=8, unit=Deadline.UNITS.WORKDAYS)
+        self.assertEqual(deadline.deadline_date, naive_date(u'2010-10-20'))
+
+    def test_calendar_days_passed_property_and_calendar_days_passed_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), _today=naive_date(u'2010-10-10'))
+        self.assertEqual(deadline.calendar_days_passed, 5)
+        self.assertEqual(deadline.calendar_days_passed_at(naive_date(u'2010-10-10')), 5)
+
+    def test_workdays_passed_property_and_workdays_passed_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), _today=naive_date(u'2010-10-10'))
+        self.assertEqual(deadline.workdays_passed, 3)
+        self.assertEqual(deadline.workdays_passed_at(naive_date(u'2010-10-10')), 3)
+
+    def test_calendar_days_remaining_property_and_calendar_days_remaining_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-08'))
+        self.assertEqual(deadline.calendar_days_remaining, 5)
+        self.assertEqual(deadline.calendar_days_remaining_at(naive_date(u'2010-10-08')), 5)
+
+    def test_workdays_remaining_property_and_workdays_remaining_at_method(self):
+        deadline =  self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-08'))
+        self.assertEqual(deadline.workdays_remaining, 3)
+        self.assertEqual(deadline.workdays_remaining_at(naive_date(u'2010-10-08')), 3)
+
+    def test_calendar_days_behind_property_and_calendar_days_behind_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-26'))
+        self.assertEqual(deadline.calendar_days_behind, 13)
+        self.assertEqual(deadline.calendar_days_behind_at(naive_date(u'2010-10-26')), 13)
```

Comment:
@viliambalaz Pri `foo_at` metodach by si mohol pouzit iny datum, nez pri `foo` properties. Nech sa 2x netestuje to iste, a nech sa overi, ze `foo_at` to vypocita vzhladom na zadany datum. Idealne pouzit taky datum, ze aj vysledok `foo` a `foo_at` je rozny.

Reply:
:ok: Test `test_is_deadline_missed_property_and_is_deadline_missed_at_method` splna poziadavku, pretoze sa testuje aj stav `True` aj stav `False` pri 2 roznych datumoch.